src/runtime/metatype.cs

```diff
@@ -288,8 +299,10 @@ public static void tp_dealloc(IntPtr tp)
             var flags = (TypeFlags)Util.ReadCLong(tp, TypeOffset.tp_flags);
             if ((flags & TypeFlags.Subclass) == 0)
             {
-                IntPtr gc = Marshal.ReadIntPtr(tp, TypeOffset.magic());
-                ((GCHandle)gc).Free();
+                GetGCHandle(new BorrowedReference(tp)).Free();
+#if DEBUG
+                SetGCHandle(new BorrowedReference(tp), Runtime.CLRMetaType, default);
```

Comment:
It is a legitimate question, and since took me a while to understand why I put it here and then forgot.

What it does in Debug build is that instead of CLR crash (typically `ExecutionEngineException`) you might get when accessing bad `GCHandle` it raises a debuggable exception, that you might notice when running tests in debug build.

Reply:
To never forget again, a comment would be good.